packages/wonder-blocks-dropdown/components/multi-select-menu.js

```diff
@@ -0,0 +1,227 @@
+            .map((item) => {
+                return {
+                    value: item.value,
+                    selected: true,
+                };
+            });
+        onChange(update);
+    }
+
+    handleSelectNone() {
+        const {items, onChange, selectedItems} = this.props;
+        const update = items
+            .filter((item) => selectedItems.includes(item.value))
+            .map((item) => {
+                return {
+                    value: item.value,
+                    selected: false,
+                };
+            });
+        onChange(update);
+    }
+
+    render() {
+        const {
+            alignment,
+            displaySelectionShortcuts,
+            disabled,
+            items,
+            light,
+            placeholder,
+            selectItemType,
+            selectedItems,
+            style,
+        } = this.props;
+
+        const {open} = this.state;
+
+        // TODO(sophie): figure out how to configure plurals for i18n
+        const menuText =
+            selectedItems.length === 0 && placeholder
+                ? placeholder
+                : `${selectedItems.length} ${selectItemType}`;
+
+        const opener = (
+            <SelectBox
+                disabled={disabled}
+                light={light}
+                onClick={() => this.toggleMenu()}
+                style={style}
+            >
+                {menuText}
+            </SelectBox>
+        );
+
+        const menuItems = items.map((item, index) => {
+            return (
+                <SelectItem
+                    disabled={item.disabled}
+                    key={item.value}
+                    label={item.label}
+                    onClick={item.onClick}
+                    onToggle={(value, state) =>
+                        this.handleSelected(value, state)
+                    }
+                    selected={selectedItems.includes(item.value)}
+                    value={item.value}
+                    variant={"checkbox"}
+                />
+            );
+        });
+
+        if (displaySelectionShortcuts) {
+            // TODO(sophie): translate for i18n
+            const selectAll = (
+                <ActionItem
+                    label={`Select all (${items.length})`}
+                    disabled={items.length === selectedItems.length}
+                    indent={true}
+                    onClick={() => this.handleSelectAll()}
+                />
+            );
+
+            const selectNone = (
+                <ActionItem
+                    label={"Select none"}
+                    disabled={selectedItems.length === 0}
+                    indent={true}
+                    onClick={() => this.handleSelectNone()}
+                />
+            );
+
+            const separator = <SeparatorItem key={"selectOptSep"} />;
+
+            // Add options at beginning, add a separator
+            menuItems.unshift([selectAll, selectNone, separator]);
```

Comment:
Maybe move the `displaySelectionShortcuts` block above the call to `const menuItems = items.map((item, index) => {`.  This avoids `unshift()` which can be bad for perf... in this situation not a huge deal.  Maybe something like:
```
const menuItems = [...this.maybeGetShortcuts(), ...this.getMenuitems()];
```